### DISCUSSION TITLE
Getting Rhino to build with htmlunit

### DISCUSSION THREAD

**Discussion by sengsational** [mozilla/rhino#1228] [Q&A] [answered]
In the IntelliJ IDE (2022.1), I have been able to get HtmlUnit (https://github.com/HtmlUnit/htmlunit) to build and run.  I've also got a dependency, neko (https://github.com/HtmlUnit/htmlunit-neko), building from source.  Both of these are Maven projects that are in their own IntelliJ project.  I made a new project, and added those two projects, which build from source, so my changes become active.

But when I try to add Rhino [actually a forked version (https://github.com/HtmlUnit/htmlunit-rhino-fork)], I'm unable to get it integrated such that I can alter the Rhino code and have that be compiled and run my changes.  The rhino-fork code runs, but runs from the dependency specified in the Maven build, so I'm unable to change the source and have it build and run.

So this is a IDE/build question where I could use very specific PhD (push here, dummy) instructions for getting rhino integrated into this build such that I can change rhino source and test the result.

**Comment by rbri** [comment, marked as answer]
Hi @sengsational,
sorry for being late on your question.
I think this is more for HtmlUnit maybe we have to discuss it there in more detail.

The general idea: the there are two project core-js and rhino-fork.
The rhino-fork is a slightly modified version of rhino. The project is there to maintain the regular sync with the rhino sources.
The core-js part is more the deployment helper - the build script uses the rhino-fork project and patches all classes into new packages. Together wish some tests and some adapter code these classes are building core-js.

How do i do development in this environment?
Usually i use the HtmlUnit project togethere with the core-js project to do all the debugging. If i have an idea what is going wrong i try to build a minimal test case that shows the problem - and usually i have an idea if the problem is in the adapter code or in rhino itself.

In most cases the problem is in Rhino - so i switch then to the rhino project itself and implement the test case there (and sometimes also the fix :-))

Hope that helps....
